frontend/public/components/catalog/catalog-page.tsx

```diff
@@ -344,6 +346,30 @@ export const CatalogListPage = withExtensions<CatalogListPageExtensionProps>({
   },
 );
 
+export const filterIncompatibleHelmCharts = (
+  charts: HelmChartEntries,
+  kubernetesVersion: string,
+): HelmChartEntries => {
+  const filteredHelmcharts: HelmChartEntries = _.reduce(
+    charts,
+    (acc, helmCharts: HelmChart[], helmChartName: string) => {
+      let arr: HelmChart[] = [];
+      _.forEach(helmCharts, (obj: HelmChart) => {
+        if (obj?.kubeVersion && semver.valid(kubernetesVersion)) {
+          if (semver.satisfies(kubernetesVersion, obj.kubeVersion)) {
+            arr = [...arr, obj];
```

Comment:
why not push? Does it have a shallow compare somewhere?

Reply:
+1 for add.push here.
And just one addition when you accept this change: Give things more meaningful names, like `const filteredCharts` instead of arr and maybe helmChart instead of just obj? :)